src/custom/state/claim/hooks/utils.ts

```diff
@@ -215,3 +217,20 @@ export function calculateInvestmentAmounts(
   const amount = CurrencyAmount.fromRawAmount(currencyAmount.currency, investedAmount)
   return { vCowAmount: price.quote(amount), investmentCost: amount }
 }
+
+/**
+ * Helper function that prepares investFlowData for claiming by calculating vCowAmount from investedAmounts
+ */
+export function prepareInvestClaims(investFlowData: InvestClaim[], userClaimData: EnhancedUserClaimData[]) {
```

Comment:
good call adding a helper function instead of complicating more the tx preparation logic